metricbeat/docs/autodiscover-hints.asciidoc

```diff
@@ -96,6 +96,28 @@ co.elastic.logs/processors.add_locale.abbrevation: "PST"
 
 In the above sample the processor definition tagged with `1` would be executed first.
 
+When hints are used along with templates, then hints will be evaluated only in case
+there is no template's condition that resolve to true. For example:
+
+[source,yaml]
+-----
+metricbeat.autodiscover.providers:
+  - type: kubernetes
+    hints.enabled: true
+    templates:
+        - condition:
+            contains:
+              kubernetes.annotations.prometheus.io/scrape: "true"
+          config:
+            - module: prometheus
+              metricsets: ["collector"]
+              hosts: "${data.host}:${data.port}"
+-----
+
+In this example first the condition `kubernetes.annotations.prometheus.io/scrape: "true"`
+is evaluated and if not matched the hints will be processed.
```

Comment:
`metricbeat` doesn't have `hints.default_config`, that is why it is omitted here?

Reply:
Yeap. See [https://github.com/elastic/beats/blob/95626b8f1690344312c0831ab2bdcbccffe4d089/metricbeat/autodiscover/builder/hints/config.go#L30](https://github.com/elastic/beats/blob/95626b8f1690344312c0831ab2bdcbccffe4d089/metricbeat/autodiscover/builder/hints/config.go#L30) compared to [https://github.com/elastic/beats/blob/661ff1412598d2da296fa6385b5751ac3d1792b9/filebeat/autodiscover/builder/hints/config.go#L39](https://github.com/elastic/beats/blob/661ff1412598d2da296fa6385b5751ac3d1792b9/filebeat/autodiscover/builder/hints/config.go#L39). In fact I don't think it would even had any reason to have a default config for metris. Logs is kind of special since all containers can provide them in a unified way.